calcom/cal.com
# chore: Make URLs for terms and privacy policy customizable via .env vars

## What does this PR do?



This PR allows self-hosted instances to set custom URLs for Terms and Privacy Policy by introducing two new `.env` vars `NEXT_PUBLIC_WEBSITE_PRIVACY_POLICY_URL` and `NEXT_PUBLIC_WEBSITE_TERMS_URL`.  The custom URLs will then be displayed on the signup page as well as on the event booking form.

- Fixes #15373
- Fixes CAL-3913


**Signup page before**
![image](https://github.com/user-attachments/assets/3a1b0ffc-89a4-4295-8fd9-5eb34250b3b2)
This was set to `${WEBSITE_URL}/terms` (note the URL in the bottom left corner of the screenshot). While this page exists for the Cal.com platform it doesn't exist in the case of self-hosted instances which just led to a HTTP 404 error. See screenshot for terms URL (same applied to the privacy policy URL).
![image](https://github.com/user-attachments/assets/de6c6fee-807a-4886-8fe6-12a743d390a6)


**Signup page after (with custom URLs)**
![image](https://github.com/user-attachments/assets/695a95a1-2077-4d1a-87b6-9849bfe1426f)
Note the link to custom terms in the bottom left corner when hovering over the terms link. Same applies to privacy policy link.


**Signup page after (without custom URLs - fall back to Cal.com terms and privacy policy)**
![image](https://github.com/user-attachments/assets/437da55a-29c3-4f5c-8f1e-6a0a7924c849)
Note the link to Cal.com's terms in the bottom left corner when hovering over the terms link. Same applies to privacy policy link.


**Event booking form before**
![image](https://github.com/user-attachments/assets/faca3116-cea0-4d5c-98d1-9299865623f2)
There was no reference to terms and privacy policy at all (was removed in https://github.com/calcom/cal.com/pull/13966).


**Event booking form after (with custom URLs)**
![image](https://github.com/user-attachments/assets/4b1789ca-dfbe-4d89-826e-e90ab5172629)
Note the link to custom terms in the bottom left corner when hovering over the terms link. Same applies to privacy policy link.


**Event booking form after (without custom URLs - fall back to Cal.com terms and privacy policy)**
![image](https://github.com/user-attachments/assets/ec9cb688-2357-4c16-a7f0-e5eada4c276a)
Note the link to Cal.com's terms in the bottom left corner when hovering over the terms link. Same applies to privacy policy link.



## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?



Set the `.env` vars `NEXT_PUBLIC_WEBSITE_PRIVACY_POLICY_URL` and `NEXT_PUBLIC_WEBSITE_TERMS_URL` to any URL. Then visit 
- Signup page, e.g. http://localhost:3000/signup
- Book event form, e.g. http://localhost:3000/{USER_NAME}/{MEETING_NAME}?month=2024-08&date=2024-08-26&slot=2024-08-26T14%3A00%3A00.000Z

On both pages hover over the `Terms`and `Privacy` links and verify that they point to the URLs you set in `.env` (`NEXT_PUBLIC_WEBSITE_PRIVACY_POLICY_URL` and `NEXT_PUBLIC_WEBSITE_TERMS_URL`).
